synapse/rest/client/v1/room.py

```diff
@@ -568,7 +568,15 @@ def __init__(self, hs):
     @defer.inlineCallbacks
     def on_GET(self, request, room_id, event_id):
         requester = yield self.auth.get_user_by_req(request, allow_guest=True)
-        event = yield self.event_handler.get_event(requester.user, room_id, event_id)
+        try:
+            event = yield self.event_handler.get_event(
+                requester.user, room_id, event_id
+            )
+        except AuthError as e:
+            # This endpoint is supposed to return a 404 when the requester does
+            # not have permission to access the event
+            # https://matrix.org/docs/spec/client_server/r0.5.0#get-matrix-client-r0-rooms-roomid-event-eventid
+            return (404, e.msg)
```

Comment:
I think this should be a `(404, "Event not found.")`, as the only reason to not return a 403 is to hide the fact the event exists?

Reply:
I'm not sure if the original intention in the spec was to hide that fact, but it does seem like a good idea.